File: frontend/src/features/form/types/question-response-dto.ts
import { AnswerResponseDto } from "@/features/form/types/answer-response-dto";
import { QuestionType } from "@/features/form/types/question-type";

export interface QuestionResponseDto {
  id: string;
  content: string;
  type: QuestionType;
  imageKey: string | null;
  isRequired: boolean;
  answers: AnswerResponseDto[] | null;
}

File: backend/src/main/java/format/backend/form/application/shared/dto/QuestionResponseDto.java
package format.backend.form.application.shared.dto;

import format.backend.form.domain.entity.QuestionType;
import java.util.List;
import org.jspecify.annotations.Nullable;

public record QuestionResponseDto(
        String id,

        String content,

        QuestionType type,

        @Nullable String image,

        boolean isRequired,

        List<AnswerResponseDto> answers) {}
